base/loading.jl

```diff
@@ -123,36 +120,28 @@ dummy_uuid(project_file::String) = isfile_casesensitive(project_file) ?
 
 ## package path slugs: turning UUID + SHA1 into a pair of 4-byte "slugs" ##
 
-const SlugInt = UInt32 # max p = 4
-const chars = String(['A':'Z'; 'a':'z'; '0':'9'])
-const nchars = SlugInt(length(chars))
-const max_p = floor(Int, log(nchars, typemax(SlugInt) >>> 8))
+const slug_chars = String(['A':'Z'; 'a':'z'; '0':'9'])
 
-function slug(x::SlugInt, p::Int)
-    1 ≤ p ≤ max_p || # otherwise previous steps are wrong
-        error("invalid slug size: $p (need 1 ≤ p ≤ $max_p)")
-    return sprint() do io
+function slug(x::UInt32, p::Int)
+    sprint(sizehint=p) do io
+        n = length(slug_chars)
         for i = 1:p
-            x, d = divrem(x, nchars)
-            write(io, chars[1+d])
+            x, d = divrem(x, n)
+            write(io, slug_chars[1+d])
         end
     end
 end
-slug(x::Integer, p::Int) = slug(SlugInt(x), p)
 
-function slug(bytes::Vector{UInt8}, p::Int)
-    n = nchars^p
-    x = zero(SlugInt)
-    for (i, b) in enumerate(bytes)
-        x = (x + b*powermod(2, 8(i-1), n)) % n
-    end
-    slug(x, p)
+function package_slug(uuid::UUID, p::Int=4)
+    crc = _crc32c(uuid)
```

Comment:
This seems like an unusual use of crc-32. Aren't you really trying to do `objectid(uuid)`, but since `uuid` is already random, its also just `uuid % UInt32`?

Reply:
If the UUID was always a cryptographically secure hash or a genuinely random value that would be fine, but it isn't always. Sometimes it can be based on your mac address and the time at generation. In such cases it's much better if a change to any uuid bit tends to change any output bit. CRC by design has that property. A cryptographically secure hash would be even better but we don't have any of those available at the point where this could be used.